src/types/web-handler.ts

```diff
@@ -4,19 +4,24 @@ import { EdgeSpecRouteBundle } from "./edge-spec"
 export type EdgeSpecRouteParams = {
   [routeParam: string]: string | string[]
 }
-export interface EdgeSpecRequestOptions {
+
+export interface EdgeSpecMiddlewareOptions {
+  headers?: Record<string, string>
```

Comment:
what is this used for?

Reply:
This is for setting defaults on the response. The equivalent of `res.headers[...] = ...`, or `req.headers[...] = ...`.

I think that `responseDefaults: { headers: ..., status: ... }` would be better, what do you think? @codetheweb

Reply:
yeah I like `responseDefaults: new Response()`

the question is then if `responseDefaults` override or merge when multiple middleware use it

Reply:
I didn't think of `responseDefaults` as being passed an actual response instance, but just basically headers and status. And most of the time it should probably just be headers

> the question is then if responseDefaults overrides or merges when multiple middleware use it

It merges/cascades for sure. If users want to "override" response they can throw the error we discussed out-of-band

Reply:
> I didn't think of responseDefaults as being passed an actual response instance, but just basically headers and status. And most of the time it should probably just be headers

that's ok with me I guess, just trying to limit the number of custom concepts on top of Request/Response

> It merges/cascades for sure. If users want to "override" response they can throw the error we discussed out-of-band

yep makes sense 👍

Reply:
> that's ok with me I guess, just trying to limit the number of custom concepts on top of Request/Response

makes sense, but I guess its a little confusing which fields in `Response` will set defaults and which won't, if you pass a whole `Response`

we can try it out for now I guess, and see if it's a complaint

Reply:
![Star Wars Hoodie GIF](https://media.tenor.com/iIMv9PUP-KAAAAAC/star-wars-hoodie.gif)

Reply:
done